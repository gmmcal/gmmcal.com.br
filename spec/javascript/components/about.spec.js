import { mount } from '@vue/test-utils'
import VueMarkdown from 'vue-markdown'
import About from '@/components/about'
import create from '../settings/utils'

describe('About', () => {
  let props, wrapper

  beforeAll(() => {
    window.gon = { about: create('about') }

    props = {
      propsData: {
        image: 'bar.jpg'
      }
    }

    wrapper = mount(About, props)
  })

  test('Component is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('Component data properties', () => {
    const expected = ['about', 'I18n']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  describe('Contains', () => {
    test('page title', () => {
      const el = wrapper.find('h2')
      expect(el.text()).toBe('About Me')
    })

    test('user name', () => {
      const el = wrapper.find('h3')
      expect(el.text()).toBe('Gustavo Cunha')
    })

    test('job title', () => {
      const el = wrapper.find('.sub-heading')
      expect(el.text()).toBe('Some title at some company')
    })

    test('user image', () => {
      expect(wrapper.contains('img[src="bar.jpg"]')).toBe(true)
    })

    test('CV link', () => {
      expect(wrapper.contains('a[href="file.pdf"]')).toBe(true)
    })
  })

  describe('Child components', () => {
    test('VueMarkdown', () => {
      const child_component = wrapper.find(VueMarkdown)
      expect(child_component.isVueInstance()).toBeTruthy()
    })
  })
})
