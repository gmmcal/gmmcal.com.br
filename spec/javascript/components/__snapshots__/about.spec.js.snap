// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<section id="about" class="about-me">
  <div class="container">
    <div class="heading">
      <h2>About Me</h2>
    </div>
    <div class="row">
      <div class="col-xs-12 col-sm-12 col-md-7 col-lg-7 pb-5">
        <h3>Gustavo Cunha</h3>
        <h4 class="sub-heading">Some title at some company</h4>
        <div class="description">
          <p>Lorem ipsum nulus dolor.</p>
        </div> <a href="file.pdf" target="_blank" rel="noopener" class="btn-download">PDF Download</a>
      </div>
      <div class="col-xs-12 col-sm-12 col-md-5 col-lg-4 offset-lg-1">
        <div class="lazy topmar"></div>
      </div>
    </div>
  </div>
</section>
`;
